lib/node/index.js

```diff
@@ -640,6 +640,7 @@ Request.prototype.redirect = function(res){
   this._redirectList.push(url);
   this.emit('redirect', res);
   this.qs = {};
+  this.qsRaw = [];
```

Comment:
This actually was a bug in the original implementation of redirection: only the key/value part of the query string was cleared (`qs`), but not the raw query string parts (`qsRaw`). This behavior surfaced because this commit now stores the original query string in `qsRaw`, and an existing unit test verifies whether the query string is cleared on redirect.